PSSwagger/PSSwagger.psm1

```diff
@@ -582,6 +583,100 @@ function New-PSSwaggerModule {
 
             }
         }
+
+        # Add extra metadata based on service type
+        if (($PowerShellCodeGen['ServiceType'] -eq 'azure') -or ($PowerShellCodeGen['ServiceType'] -eq 'azure_stack')) {
+            foreach ($entry in $PathFunctionDetails.GetEnumerator()) {
+                $hyphenIndex = $entry.Name.IndexOf("-")
+                if ($hyphenIndex -gt -1) {
+                    $commandVerb = $entry.Name.Substring(0, $hyphenIndex)
+                    # Add client-side filter metadata automatically if:
+                    #   1: If the command is a Get-* command
+                    #   2: A *_List parameter set exists
+                    #   3: A *_Get parameter set exists
+                    #   4: *_List required parameters are a subset of *_Get required parameters
+                    #   5: *_Get has a -Name parameter alias
+                    if ($commandVerb -eq 'Get') {
+                        $getParameters = @()
+                        $listParameters = $null
+                        $listOperationId = $null
+                        $getOperationId = $null
+                        $nameParameterOriginalName = $null # This ideally matches the object property name
+                        $nameParameterNormalName = $null # This is the one being switched out for -Name later
+                        foreach ($parameterSetDetails in $entry.Value.ParameterSetDetails) {
+                            if ($parameterSetDetails.OperationId.EndsWith("_Get") -and
+                            (-not ($parameterSetDetails.OperationId.StartsWith("InputObject_"))) -and
+                            (-not ($parameterSetDetails.OperationId.StartsWith("ResourceId_")))) {
+                                $getOperationId = $parameterSetDetails.OperationId
+                                foreach ($parametersDetail in $parameterSetDetails.ParameterDetails) {
+                                    foreach ($parameterDetailEntry in $parametersDetail.GetEnumerator()) {
+                                        $getParameters += $parameterDetailEntry.Value
+                                        if ($parameterDetailEntry.Value.ContainsKey('Alias')) {
+                                            if ($parameterDetailEntry.Value.Alias -eq 'Name') {
```

Comment:
Actual parameter name of a resource can be Name itself. In that case, wildcard functionality is not enabled, right?

Reply:
68afddd